### PR TITLE
Add support for Python 3.11

### PR DESCRIPTION
**Pull request by ulgens**

### CODE REVIEW

**Comment by firaskafri** [comment]
@ulgens I guess tests are failing

**Comment by yuekui** [comment]
https://github.com/graphql-python/graphene-django/pull/1368 should fix all the failed tests.

**Comment by ulgens** [comment]
@firaskafri Are these close - reopens are un purpose or do we have a bug on Github's side?

**Comment by firaskafri** [comment]
@ulgens something is wrong with tests other than the failures, closed and reopened the PR to trigger them

**Comment by ulgens** [comment]
@firaskafri Please don't 🙂 . It creates misleading notifications for anyone following the PR and doesn't make much on fixing the issue. I'll check this one the weekend.

**Comment by firaskafri** [comment]
> @firaskafri Please don't 🙂 . It creates misleading notifications for anyone following the PR and doesn't make much on fixing the issue. I'll check this one on the weekend.

The test pipeline gets stuck sometimes and that is unrelated to the the actual code failure, that was the quick way to re trigger the tests. Any suggestions to fix this?

**Comment by ulgens** [comment]
@firaskafri Failing tasks have "re-run" option on their Actions page. I believe it gives a cleaner result.
CleanShot 2022-11-28 at 01 06 14@2x

**Comment by ulgens** [comment]
Tests look a bit better now. I'll check that last failing test as soon as possible.

**Comment by firaskafri** [comment]
@ulgens test for 3.11 are failing

**Comment by firaskafri** [comment]
It seems that Schema() behavior changes when using python 3.11. I could pin point the issue in enums:

In python 3.11, schema generates this:

```
            type Query {
              pet: PetModelKind
            }

            type PetModelKind {
              id: ID!
              kind: CustomEnumKind!
            }

            enum CustomEnumKind {
              \"""Cat\"""
              CAT

              \"""Dog\"""
              DOG
            }"""
```

While in older python versions:


```
            type Query {
              pet: PetModelKind
            }

            type PetModelKind {
              id: ID!
              kind: CustomEnumKind!
            }

--->      \"""An enumeration.\"""
            enum CustomEnumKind {
              \"""Cat\"""
              CAT

              \"""Dog\"""
              DOG
            }"""
```

**Comment by firaskafri** [comment]
it seems that `enum.__doc__` behaviour changed in python 3.11 which causing [`description = description or enum.__doc__`](https://github.com/graphql-python/graphene/blob/57cbef6666e2e466808cce21b8a1769ecd3fd118/graphene/types/enum.py#L66) to fail in graphene. Hopefully1502 would be a sufficient fix.

**Review by jaw9c — APPROVED** [review]

File: graphene_django/converter.py [hunk 1]
```diff
@@ -96,7 +96,12 @@ class EnumWithDescriptionsType:
         def description(self):
             return str(named_choices_descriptions[self.name])
 
-    return_type = Enum(name, list(named_choices), type=EnumWithDescriptionsType)
+    return_type = Enum(
+        name,
+        list(named_choices),
+        type=EnumWithDescriptionsType,
+        description="An enumeration.",  # Temporary fix until https://github.com/graphql-python/graphene/pull/1502 is merged
```
Comment (kiendang) [inline graphene_django/converter.py:103]: We can do `description=__ENUM_DESCRIPTION` here with

```py
import sys
__ENUM_DESCRIPTION = "An enumeration." if (sys.version_info.major, sys.version_info.minor) == (3, 11) else None
```

up top so behaviors for other Python versions won't change.

**Comment by ulgens** [comment]
Thanks for the fix guys. I rebased the branch for clarity.

**Review by firaskafri — APPROVED** [review]

**Review by kiendang — APPROVED** [review]

**Pull request merged**